source/server/admin/admin.cc

```diff
@@ -469,9 +484,16 @@ void AdminImpl::addAllConfigToDump(envoy::admin::v3::ConfigDump& dump,
 
 absl::optional<std::pair<Http::Code, std::string>>
 AdminImpl::addResourceToDump(envoy::admin::v3::ConfigDump& dump,
-                             const absl::optional<std::string>& mask,
-                             const std::string& resource) const {
-  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+                             const absl::optional<std::string>& mask, const std::string& resource,
+                             const bool include_eds) const {
+  Envoy::Server::ConfigTracker::CbsMap callbacks_map = std::move(config_tracker_.getCallbacksMap());
+  if (include_eds) {
+    if (!server_.clusterManager().clusters().empty()) {
+      callbacks_map.emplace("endpoint", [this] { return dumpEndpointConfigs(); });
```

Comment:
Same concern here.

Reply:
Removed the `std::move`

Reply:
I don't think there will be something wrong from user's perspective. For here, it just makes a copy of the `callbacks_map` in `config_tracker_` and choose whether to add eds to it or not. It won't modify the original `callbacks_map` stored in `config_tracker_` which doesn't include eds.
Using `mask` param can exclude eds, but if you want to get all configs except eds, the `mask` need to cover all the other configs like `&mask=bootstrap,static_clusters,dynamic_clusters,static_listeners,dynamic_listeners...` . Maybe a little bit complex for user?